### DISCUSSION TITLE
J4 XML Field Showon and Required Set to True

### DISCUSSION THREAD

**Discussion by frogydiak** [joomla/joomla-cms#39211] [General]
### Steps to reproduce the issue
Create an xml field and set the shownon attribute to whatever is the value of another field then set the attribute to "required=true". You'll get an error while saving that it cannot be saved because a required data is missing or something like that.

### Expected result
Shouldn't it check for "shownon"? If set and not enabled, bypass the field required validation and if enabled then it validate for required value?

### Actual result
The validation of required field does not check the field showon attribute value.

### Additional comments
I don't know if this is a bug, but looking at it as in an end user perspective it is a bug. First thing that comes to my minds is "That field is not enabled, why do I have to enter a value?" .

**Comment by chmst** [comment]
You can define a default value for this field, which is used if the field is not visible on the screen. 
I don't remember if ther are reasons for this - but looks like a valid request for me. 


<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37454">issues.joomla.org/tracker/joomla-cms/37454</a>.</sub>

**Comment by joomdonation** [comment]
As of right now, shownon is only used to UI to show/hide fields depends on certain conditions. We do not use it for validating submit form data. Through the years, I have seen several reports like this but we never work on it further. 

For the time being, I guess you will have to use default value as suggested by @chmst as a workaround. I'm converting this to discussion to see if we want to work further on it.

**Reply by crystalenka** [reply]
Converting back to issue because otherwise no one will see it to work on it.
